internal/extsvc/gitlab/client.go

```diff
@@ -172,24 +172,22 @@ type Client struct {
 	// The URN of the external service that the client is derived from.
 	urn string
 
-	baseURL          *url.URL
-	httpClient       httpcli.Doer
-	projCache        *rcache.Cache
-	Auth             auth.Authenticator
-	rateLimitMonitor *ratelimit.Monitor
-	rateLimiter      *ratelimit.InstrumentedLimiter // Our internal rate limiter
+	baseURL             *url.URL
+	httpClient          httpcli.Doer
+	projCache           *rcache.Cache
+	Auth                auth.Authenticator
+	externalRateLimiter *ratelimit.Monitor
+	internalRateLimiter *ratelimit.InstrumentedLimiter // Our internal rate limiter
+	waitForRateLimit    bool
+	numRateLimitRetries int
```

Comment:
```suggestion
	maxRateLimitRetries int
```

It doesn't count the number of current retries, right? But rather it's the upper limit.